neqo-http3/tests/httpconn.rs

```diff
@@ -111,6 +112,38 @@ fn connect_peers(hconn_c: &mut Http3Client, hconn_s: &mut Http3Server) -> Option
     out.dgram()
 }
 
+fn connect_peers_with_rtt(
+    hconn_c: &mut Http3Client,
+    hconn_s: &mut Http3Server,
+    net_delay: u64,
+) -> (Option<Datagram>, Instant) {
+    let net_delay = Duration::from_millis(net_delay);
+    assert_eq!(hconn_c.state(), Http3State::Initializing);
+    let mut now = now();
+    let out = hconn_c.process(None, now); // Initial
+    now += net_delay;
+    let out = hconn_s.process(out.dgram(), now); // Initial + Handshake
+    now += net_delay;
+    let out = hconn_c.process(out.dgram(), now); // ACK
+    now += net_delay;
+    mem::drop(hconn_s.process(out.dgram(), now)); //consume ACK
+    let authentication_needed = |e| matches!(e, Http3ClientEvent::AuthenticationNeeded);
+    assert!(hconn_c.events().any(authentication_needed));
+    now += net_delay;
+    hconn_c.authenticated(AuthenticationStatus::Ok, now);
+    let out = hconn_c.process(None, now); // Handshake
+    assert_eq!(hconn_c.state(), Http3State::Connected);
+    now += net_delay;
+    let out = hconn_s.process(out.dgram(), now); // Handshake
+    now += net_delay;
+    let out = hconn_c.process(out.dgram(), now);
+    now += net_delay;
+    let out = hconn_s.process(out.dgram(), now);
+    now += net_delay;
+    let out = hconn_c.process(out.dgram(), now);
```

Comment:
What do all these other ones do?  HANDSHAKE_DONE?  Control streams?

Reply:
I will add a description, but yes HANDSHAKE_DONE and control and qpack streams.